frontend/packages/dev-console/src/components/topology2/topology-utils.tsx

```diff
@@ -0,0 +1,155 @@
+import * as React from 'react';
+import { EdgeModel, Edge, Model, NodeModel, Node, NodeShape } from '@console/topology';
+import { confirmModal, errorModal } from '@console/internal/components/modals';
+import { TopologyDataModel } from '../topology/topology-types';
+import {
+  createTopologyResourceConnection,
+  removeTopologyResourceConnection,
+  updateTopologyResourceApplication,
+} from '../topology/topology-utils';
+import { TYPE_APPLICATION_GROUP } from './consts';
+
+const topologyModelFromDataModel = (dataModel: TopologyDataModel): Model => {
+  const nodes: NodeModel[] = dataModel.graph.nodes.map((d) => {
+    if (d.type === 'knative-service') {
+      return {
+        width: 104,
+        height: 104,
```

Comment:
I'll reiterate my uncomfortableness with this magic number for the topology stuff. `104` isn't descriptive, especially when used in equations (at other points in the code).